### DISCUSSION TITLE
"That" identified as having a dependency of "dobj"

### DISCUSSION THREAD

**Discussion by johnosbb** [explosion/spaCy#13163] [Help: Other Questions] [answered]
I am a little confused by how SpaCy parses the following sentence:

"It was a dog that she killed."

| Text         | Index  | POS      | Tag      | Dep      | Dep Detail               | Ancestors            | Children   | Token Head   | Sub Tree     |
| ------ | ------ | ---- | ------- | ------- | --------- |  ------- | ------- | ------- | ------- |
| It           | 0      | PRON     | PRP      | nsubj    | nominal subject          | was                  |            |  was          |  It           |
| was          | 1      | AUX      | VBD      | ROOT     | root                     |                      | It dog .   |  was          |  It was a dog that she killed . |
| a            | 2      | DET      | DT       | det      | determiner               | dog was              |            |  dog          |  a            |
| dog          | 3      | NOUN     | NN       | attr     | attribute                | was                  | a killed   |  was          |  a dog that she killed |
| that         | 4      | PRON     | WDT      | dobj     | direct object            | killed dog was       |            |  killed       |  that         |
| she          | 5      | PRON     | PRP      | nsubj    | nominal subject          | killed dog was       |            |  killed       |  she          |
| killed       | 6      | VERB     | VBD      | relcl    | relative clause modifier | dog was              | that she   |  dog          |  that she killed |
| .            | 7      | PUNCT    | .        | punct    | punctuation              | was                  |            |  was          |  .            |

The sentence is in the active voice with "dog" being a subject compliment. Why does the Relative Pronoun "that" have a dependency of "dobj"?

**Comment by adrianeboyd** [comment, marked as answer]
It might be easier to see this from the parse tree:

![Screenshot_2023-11-29_20-58-23](https://github.com/explosion/spaCy/assets/5794899/cbedb834-9c58-4284-9022-672222df4329)

It's `dobj` because the relative pronoun is the direct object of "killed".

**Reply by johnosbb** [reply]
Thank you Adriane. The diagram does help. I think I understand the logic a little better now.

"It" is the subject of the sentence.
"was" is a linking verb that connects the subject "it" to the subject complement "a dog". This tells us that "it" equals "a dog."
"a dog" is the subject complement and the noun being referred to.
"that" introduces a relative clause, providing more information about the noun "dog."
"she killed" is the relative clause, where "she" is the subject and "killed" is the verb.
In the relative clause "that she killed." "that" is serving as a pronoun representing the noun "dog" from the main clause. **In the context of the relative clause**, "that" is the direct object (dobj) of the verb "killed." It is the thing that the subject "she" is performing the action of killing on.
I think I became too focused on the word to word dependencies and lost sight of the overall relationships of the clauses. The diagram really does help emphasise the relationship. 
Thank you again. Your help is much appreciated.
